darwin/dataset/local_dataset.py

```diff
@@ -169,6 +173,93 @@ def parse_json(self, index: int):
+
+        Returns
+        -------
+        mean : ndarray[double]
+            Mean value (for each channel) of all pixels of the images in the input folder
+        std : ndarray[double]
+            Standard deviation (for each channel) of all pixels of the images in the input folder
+        """
+        if multi_threaded:
+            # Set up a pool of workers
+            with mp.Pool(mp.cpu_count()) as pool:
+                # Online mean
+                results = pool.map(self._return_mean, self.images_path)
+                mean = np.sum(np.array(results), axis=0) / len(self.images_path)
+                # Online image_classification deviation
+                results = pool.starmap(self._return_std, [[item, mean] for item in self.images_path])
+                std_sum = np.sum(np.array([item[0] for item in results]), axis=0)
+                total_pixel_count = np.sum(np.array([item[1] for item in results]))
+                std = np.sqrt(std_sum / total_pixel_count)
+                # Shut down the pool
+                pool.close()
+                pool.join()
+            return mean, std
+        else:
+            # Online mean
+            results = [self._return_mean(f) for f in self.images_path]
+            mean = np.sum(np.array(results), axis=0) / len(self.images_path)
+            # Online image_classification deviation
+            results = [self._return_std(f, mean) for f in self.images_path]
+            std_sum = np.sum(np.array([item[0] for item in results]), axis=0)
+            total_pixel_count = np.sum(np.array([item[1] for item in results]))
+            std = np.sqrt(std_sum / total_pixel_count)
+            return mean, std
+
+    def measure_weights(self, **kwargs):
+        """Computes the class balancing weights (not the frequencies!!) given the train loader
+
+        Returns
+        -------
+        class_weights : ndarray[double]
+            Weight for each class in the train set (one for each class)
+        """
+        raise NotImplementedError("Base class Dataset does not have an implementation for this")
+
+    @staticmethod
+    def _compute_weights(labels: Collection):
+        """Given an array of labels computes the weights normalized
+
+        Parameters
+        ----------
+        labels : ndarray[int]
+            Array of labels
+
+        Returns
+        -------
+        ndarray[float]
+            Array of weights (one for each unique class) which are the inverse of their frequency
+        """
+        class_support = np.unique(labels, return_counts=True)[1]
+        class_frequencies = class_support / len(labels)
+        # Class weights are the inverse of the class frequencies
+        class_weights = 1 / class_frequencies
+        # Normalize vector to sum up to 1.0 (in case the Loss function does not do it)
+        class_weights /= class_weights.sum()
+        return class_weights
+
+    # Loads an image with OpenCV and returns the channel wise means of the image.
```

Comment:
Correct me if I'm wrong but we aren't using OpenCV, right?

Reply:
Yeah, we're using Pillow now. Forgot to update the comment.